src/conshdlr.jl

```diff
@@ -0,0 +1,125 @@
+# Current limitations:
+# - use fixed values for some properties
+#   - EAGERFREQ = 100
+#   - NEEDSCONS = TRUE
+#   - SEPAPRIORITY = 0
+#   - SEPAFREQ = -1
+#   - DELAYSEPA = FALSE
+#   - PROPFREQ = -1
+#   - DELAYPROP = -1
+#   - PROP_TIMING = SCIP_PROPTIMING_BEFORELP
+#   - PRESOLTIMING = SCIP_PRESOLTIMING_MEDIUM
+#   - MAXPREROUNDS = -1
+# - don't support optional methods: CONSHLDRCOPY, CONSFREE, CONSINIT, CONSEXIT,
+#   CONSINITPRE, CONSEXITPRE, CONSINITSOL, CONSEXITSOL, CONSDELETE, CONSTRANS,
+#   CONSINITLP, CONSSEPALP, CONSSEPASOL, CONSENFORELAX, CONSPROP, CONSPRESOL,
+#   CONSRESPROP, CONSACTIVE, CONSDEACTIVE, CONSENABLE, CONSDISABLE, CONSDELVARS,
+#   CONSPRINT, CONSCOPY, CONSPARSE, CONSGETVARS, CONSGETNVARS, CONSGETDIVEBDCHGS
+# - don't support linear or nonlinear constraint upgrading
+
+# Abstract Types:
+
+# (also: CONSHDLRDATA)
+abstract type AbstractConstraintHandler end
+
+# (also: CONSDATA)
+abstract type AbstractConstraint{Handler} end
+
+# Example implementation
+mutable struct ExampleConstraintHandler <: AbstractConstraintHandler
+end
+
+### ^ above: what it will look like to Julia users
+### = here: sketches for the translation
+
+### v below: what it will look like to SCIP
+
+# Methods
+
+# CONSENFOLP
+# (SCIP* scip, SCIP_CONSHDLR* conshdlr, SCIP_CONS** conss, int nconss,
+#  int nusefulconss, SCIP_Bool solinfeasible, SCIP_RESULT* result)
+
+function _consenfolp(scip::Ptr{SCIP_}, conshdlr::Ptr{SCIP_CONSHDLR},
+                     conss::Ptr{Ptr{SCIP_CONS}}, nconss::Cint,
+                     nusefulconss::Cint, solinfeasible::SCIP_Bool,
+                     result::Ptr{SCIP_RESULT})
+    return SCIP_OKAY
```

Comment:
Can you redirect to the user-provided callback by storing it in SCIP_CONSDATA?

Reply:
Yes, that might work. Then we would have setter methods to store functions in the `AbstractConstraintHandler` subtype, I guess. Some of these are required, so it might be better to set them in the constructor already.